util/fipstools/inject_hash_cpp/inject_hash.cpp

```diff
@@ -0,0 +1,61 @@
+#include <LIEF/LIEF.hpp>
+#include <iostream>
+#include <string>
+#include <cstring>
+
+static void print_usage() {
+    std::cout << "Usage: inject_hash_cpp -o output_file -in-object input_file [-apple]" << std::endl;
+}
+
+int main(int argc, char** argv) {
+    std::cout << "\n=== C++ inject_hash starting ===" << std::endl;
+
+    // Parse command line arguments
+    const char* input_file = nullptr;
+    const char* output_file = nullptr;
+    bool is_apple = false;
```

Comment:
Is it possible to re-use any of our existing argument parsing from our tools?

Reply:
I also brought this up in https://github.com/aws/aws-lc/pull/2534#discussion_r2202011937. The answer seemed to imply that this would part of a larger future refactor/PR?